### PR TITLE
handle noop correctly

### PR DESCRIPTION
**Pull request by mattspataro**
We recently migrated our codebase from OpenTracing to OpenTelemetry and were delighted to take advantage of the `OpenTracingShim`. Unfortunately we ran into issues with noop spans.

There are a few instances in which we register new spans as children of a `io.opentracing.noop.NoopSpan`. Unfortunately, when this happens, the `OpenTelemetryShim` attempts to turn the parent `NoopSpan` into a `SpanShim`, which throws an exception. This issue can be mitigated by performing a noop if we attempt to register a `NoopSpan` as a parent. It appears that this was meant to be handled anyways, judging from the following TODO comment that already existed in the codebase:
`// TODO - Verify we handle a no-op Span`

This PR fixes this issue and provides a unit test to verify that it works correctly.

### CODE REVIEW

**Review by tmccombs — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

File: opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/SpanBuilderShimTest.java [hunk 1]
```diff
@@ -59,6 +61,19 @@ void parent_single() {
...
   }
 
+  @Test
+  void noop_parent_span() {
+    SpanShim childSpan =
+        (SpanShim)
+            new SpanBuilderShim(telemetryInfo, SPAN_NAME).asChildOf(NoopSpan.INSTANCE).start();
+    try {
+      SpanData spanData = ((ReadableSpan) childSpan.getSpan()).toSpanData();
+      assertThat(spanData.getParentSpanId()).isEqualTo(SpanId.getInvalid());
```
Comment (sriraamas) [inline opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/SpanBuilderShimTest.java:71]: nit.. may be 
```
assertThat(SpanId.isValid(spanData.getParentSpanId())).isFalse();
```

**Review by sriraamas — APPROVED** [review]

**Comment by jkwatson** [comment]
@carlosalberto can you take a look at this one?

File: opentracing-shim/build.gradle.kts [hunk 2]
```diff
@@ -11,11 +11,15 @@ dependencies {
...
   implementation(project(":semconv"))
+  implementation("io.opentracing:opentracing-noop:0.33.0")
 
   annotationProcessor("com.google.auto.value:auto-value")
 
   testImplementation(project(":sdk:testing"))
 }
+repositories {
+  mavenCentral()
+}
```
Comment (jack-berg) [inline opentracing-shim/build.gradle.kts:22]: This isn't needed

File: opentracing-shim/build.gradle.kts [hunk 3]
```diff
@@ -11,11 +11,15 @@ dependencies {
 
   api("io.opentracing:opentracing-api")
   implementation(project(":semconv"))
+  implementation("io.opentracing:opentracing-noop:0.33.0")
```
Comment (jack-berg) [inline opentracing-shim/build.gradle.kts:14]: We centrally manage dependency version in [./dependencyManagement/build.gradle.kts](https://github.com/open-telemetry/opentelemetry-java/blob/main/dependencyManagement/build.gradle.kts#L77). Can you add this entry to the list?

**Comment by carlosalberto** [comment]
So this is an undocumented/corner-case of the OpenTracing API - for safety reasons, though, I'm happy to accept it - (I will add a note in the Spec's section for this, so other languages support this scenario as well).

(OpenTracing's No-op `Span` is _not quite_ the same as our OpenTelemetry's `Span.getInvalid()`)

**Review by carlosalberto — APPROVED** [review]

File: opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/SpanBuilderShimTest.java (hunk 1 above)
Comment (mattspataro) [reply opentracing-shim/src/test/java/io/opentelemetry/opentracingshim/SpanBuilderShimTest.java:71]: done

**Comment by mattspataro** [comment]
> So this is an undocumented/corner-case of the OpenTracing API - for safety reasons, though, I'm happy to accept it - (I will add a note in the Spec's section for this, so other languages support this scenario as well).
> 
> (OpenTracing's No-op `Span` is _not quite_ the same as our OpenTelemetry's `Span.getInvalid()`)

I realized later that the java doc for `io.opentracing.Tracer.SpanBuilder.asChildOf` says: "If parent==null, this is a noop." So our code-base should've been passing in `null` for noop spans anyways. But still, I think this could be helpful for those that make the same mistake as us and accidentally pass in a noop span.

**Review by jack-berg — APPROVED** [review]

**Pull request merged**